### PR TITLE
Fix two problem

### PR DESCRIPTION
**Pull request by ziben**
When I use the extension, I found tow problems.

First, after the first format, the separator contains multiple '-', so the continues format failed.
Second, when the table contains CJK chars, such as chinese, the cellLength is correct.

### CODE REVIEW

File: src/column.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ export class ColumnFactory {
             colCount--;
         } else if (colCount > 1 && rawColumns[0].isEmpty() && !rawColumns[colCount - 1].isEmpty()) {
             // add an empty column at the end if the first one is an empty column but the last one isn't
-            const emptyRawRows = new Array(rowCount).fill(0);
+            const emptyRawRows = new Array(rowCount).fill('');
```
Comment (darkriszty) [inline src/column.ts]: Why was this changed?

File: src/column.ts [hunk 2]
```diff
@@ -120,21 +120,27 @@ export class ColumnFactory {
...
+    public cellLengths: number[] = [];
+    constructor(public columnValues: string[]) {
         this._setCellLength();
     }
 
     isEmpty(): boolean {
         return this.cellLength == 0;
     }
 
+    getLenth(columnValue: string): number {
```
Comment (darkriszty) [inline src/column.ts]: This method should be private and prefixed with underscore.

File: src/column.ts [hunk 3]
```diff
@@ -120,21 +120,27 @@ export class ColumnFactory {
 
 export class RawColumn {
     public cellLength: number = 0;
-    constructor(public columnValues: string[]) { 
+    public cellLengths: number[] = [];
```
Comment (darkriszty) [inline src/column.ts]: There shouldn't be an additional array to do this job, I think that cellLength could be used.

File: src/column.ts [hunk 4]
```diff
@@ -120,21 +120,27 @@ export class ColumnFactory {
...
+    constructor(public columnValues: string[]) {
         this._setCellLength();
     }
 
     isEmpty(): boolean {
         return this.cellLength == 0;
     }
 
+    getLenth(columnValue: string): number {
+        return columnValue.replace(/[^\x00-\xff]/g, "aa").length; // 双字节长度为2
```
Comment (darkriszty) [inline src/column.ts]: Why just "x00-\xff" ? Why not more?

Comment (darkriszty) [inline src/column.ts]: Replacing it with "aa" is just a hack. It should be tested separately to return a given length.

Comment (darkriszty) [inline src/column.ts]: Please don't use non-english comments.

File: src/column.ts [hunk 5]
```diff
@@ -120,21 +120,27 @@ export class ColumnFactory {
 
 export class RawColumn {
```
Comment (darkriszty) [inline src/column.ts]: There should be some unit tests added for this class to cover the new functionality.

File: src/tableFactory.ts [hunk 6]
```diff
@@ -35,19 +35,24 @@ class TableValidator {
     }
 
     private static isSeparator(cellValue: string, isFirstOrLast: boolean): boolean {
-        if (cellValue.trim() == "-")
-            return true;
+        // if (cellValue.trim() == "-")
```
Comment (darkriszty) [inline src/tableFactory.ts]: Please don't leave commented code hanging around. In case history is needed, there is source control for it.

**Comment by darkriszty** [comment]
Hi, and thank you for taking the time to contribute. I wasn't expecting contributions so soon :)

I already fixed the first issue with the single dash problem, as well as another problem related to the separator detection, so there is no need to merge your first commit.
For your second commit, I created an issue on the issues page. I made a little research and maybe we can discuss it further to come to an agreement on it, so at the moment I would not like to merge your second commit either.
Regardless of the fact that I won't merge your PR now, I actually made a review so you know what to look out for in the future.

Also for future reference, please use one Pull Request to solve one issue from the issues page. I added a short contribution guide to the master branch.

**Comment by ziben** [comment]
Ok , thanks for the review.

File: src/column.ts (hunk 3 above)
Comment (ziben) [inline src/column.ts]: Becuase in other place need to compute the every col's length, so I cached in there.

File: src/column.ts (hunk 4 above)
Comment (ziben) [inline src/column.ts]: The "x00-\xff"  is to match the char who's unicode is out of range 0-255.

Comment (ziben) [inline src/column.ts]: Yes, should add some test!

Comment (ziben) [inline src/column.ts]: Sorry about the comment. I'll use english comments next time.

File: src/column.ts (hunk 1 above)
Comment (ziben) [inline src/column.ts]: Because the replace method can only apply to string.
